tests/handlers/test_room_summary.py

```diff
@@ -316,6 +318,59 @@ def test_visibility(self):
             AuthError,
         )
 
+    def test_room_hierarchy_cache(self) -> None:
+        """In-flight room hierarchy requests are deduplicated."""
+        # Run two `get_room_hierarchy` calls up until they block.
```

Comment:
"up until they block" is why we manually call `ensureDeferred` instead of letting `get_success` do that?

Reply:
Yep, otherwise they'd run sequentially and the cache wouldn't do anything.

Reply:
That's what I thought, just double checking!